File: backend/app/schemas/securite.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class AuditLogResponse(BaseModel):
    id: UUID
    user_id: UUID | None
    user_email: str | None
    action: str
    resource_type: str
    resource_id: str | None
    details: str | None
    ip_address: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class HistoriqueNoteResponse(BaseModel):
    id: UUID
    evaluation_id: UUID
    eleve_id: UUID
    annee_scolaire_id: UUID
    ancienne_valeur: str | None
    nouvelle_valeur: str | None
    ancien_absent: bool | None
    nouveau_absent: bool | None
    modifie_par_id: UUID | None
    ip_address: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class HistoriquePaiementResponse(BaseModel):
    id: UUID
    paiement_id: UUID
    eleve_id: UUID
    annee_scolaire_id: UUID
    action: str
    montant: str
    statut_avant: str | None
    statut_apres: str
    details: str | None
    modifie_par_id: UUID | None
    ip_address: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class SauvegardeStatusResponse(BaseModel):
    repertoire: str
    derniere_sauvegarde: str | None
    taille_octets: int | None
    chiffrement: str
    procedure_restauration: str
    https_requis: bool

File: frontend/src/types/securite.ts
export type AuditLog = {
  id: string;
  user_id: string | null;
  user_email: string | null;
  action: string;
  resource_type: string;
  resource_id: string | null;
  details: string | null;
  ip_address: string | null;
  created_at: string;
};

export type HistoriqueNote = {
  id: string;
  evaluation_id: string;
  eleve_id: string;
  ancienne_valeur: string | null;
  nouvelle_valeur: string | null;
  created_at: string;
};

export type HistoriquePaiement = {
  id: string;
  paiement_id: string;
  eleve_id: string;
  action: string;
  montant: string;
  statut_avant: string | null;
  statut_apres: string;
  details: string | null;
  created_at: string;
};

export type LoginLog = {
  id: string;
  email: string;
  ip_address: string | null;
  success: boolean;
  created_at: string;
};

export type SauvegardeStatus = {
  repertoire: string;
  derniere_sauvegarde: string | null;
  taille_octets: number | null;
  chiffrement: string;
  procedure_restauration: string;
  https_requis: boolean;
};
